07-routing-gossip.md

```diff
@@ -698,10 +698,6 @@ The sender:
   that it wants the gossip to refer to.
 
 The receiver:
-  - SHOULD send all gossip messages whose `timestamp` is greater or
-    equal to `first_timestamp`, and less than `first_timestamp` plus
-    `timestamp_range`.
-	- MAY wait for the next outgoing gossip flush to send these.
```

Comment:
> SHOULD send all gossip messages whose `timestamp` is greater or	
    equal to `first_timestamp`, and less than `first_timestamp` plus	
    `timestamp_range`.

This still seems like something we'd want to keep. Perhaps it's better to note that implementations should be aware of the consequences of setting `first_timestamp` to a value too far in the past?

Reply:
> This still seems like something we'd want to keep. 

Why? The whole point of moving to gossip_queries was to avoid dumping the full channel graph on every connection, and you can't really make use of this without triggering it. Maybe it should be changed to MAY rather than SHOULD?

> Perhaps it's better to note that implementations should be aware of the consequences of setting first_timestamp to a value too far in the past?

We want to be able to set timestamps in the past, that way we catch new updates arriving at tip that may be slow to propagate due to the staggered gossip. You could say that then we shouldn't set it too far back, but how do we arrive at that value.

It becomes harder to pick a value when some nodes send historical traffic and others don't. You have to choose a balance between DOSing the remote node and forcing yourself to filter out a mound of gossip data, or just accept the loss and miss out on slow-to-propagate updates.

Reply:
Also with `extended_gossip_queries` I think it makes this functionality redundant, and we should encourage people to get missing data through that rather dumping the graph

Reply:
If you really want a complete gossip blast, you can always reconnect.  Generally asking for everything to be re-transmitted is a bit antisocial.